README.md

```diff
@@ -29,13 +37,7 @@ import invariant from 'tiny-invariant';
 invariant(value, () => getExpensiveMessage());
 ```
 
-## Why `tiny-invariant`?
```

Comment:
I did like having this at the top as it called out "why" the library exists. Do you think it's not needed?

Reply:
Good point, I do think it's useful to say why. I added back the why section and gave it a couple sentences, but you can probably articulate it better than I can!